compute_sdk/tests/unit/test_client.py

```diff
@@ -331,6 +330,48 @@ def funk():
     assert func_data.metadata is None
 
 
+@pytest.mark.parametrize("dep_arg", ["searchable", "function_name"])
+def test_register_function_deprecated_args(gcc, dep_arg):
+    gcc.web_client = mock.MagicMock()
+
+    with pytest.deprecated_call() as pyt_wrn:
+        gcc.register_function(funk, **{dep_arg: "foo"})
+
+    warning = pyt_wrn.pop(DeprecationWarning)
+    assert "deprecated" in str(warning).lower()
+    assert dep_arg in str(warning)
+
+
+@pytest.mark.parametrize("function", [funk, None])
+@pytest.mark.parametrize("function_name", ["name", None])
+@pytest.mark.parametrize("function_code", ["code", None])
+def test_function_registration_data_function_name_and_code(
+    function, function_name, function_code
+):
+    if function:
+        expect_err = function_name or function_code
+    else:
+        expect_err = not (function_name and function_code)
+
+    if expect_err:
```

Comment:
I haven't thought it through at this time, but I wonder if you were to apply a similar thought-process as I presented earlier, if this test function could be made simpler, or split into to two smaller (more targeted) functions.  (The smell for me is the big "if/else" block for what to assert; not always avoidable, but also a little smelly.)

Reply:
split the test into a bunch of smaller tests